plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -234,6 +262,130 @@ public void onFailure(Exception e) {
+                    );
+            }
+
+        }, e -> {
+            if (e instanceof IndexNotFoundException) {
+                actionListener.onResponse(true);
+                return;
+            }
+            log.error("Failed to delete ML Model: " + modelId, e);
+            actionListener.onFailure(e);
+
+        }));
+    }
+
+    private void checkIngestPipelineBeforeDeleteModel(String modelId, ActionListener<Boolean> actionListener) {
+        GetPipelineRequest getPipelineRequest = new GetPipelineRequest();
+        client.execute(GetPipelineAction.INSTANCE, getPipelineRequest, ActionListener.wrap(ingestPipelineResponse -> {
+            if (!isPipelineContainsModel(
+                ingestPipelineResponse.pipelines(),
+                modelId,
+                org.opensearch.ingest.PipelineConfiguration::getConfigAsMap
+            )) {
+                actionListener.onResponse(true);
+            } else {
+                List<String> searchPipelineIds = getAllPipelineIds(
+                    ingestPipelineResponse.pipelines(),
+                    org.opensearch.ingest.PipelineConfiguration::getId
+                );
+                actionListener
+                    .onFailure(
+                        new OpenSearchStatusException(
+                            searchPipelineIds.size()
+                                + " ingest pipelines are still using this model, please delete or update the pipelines first: "
+                                + Arrays.toString(searchPipelineIds.toArray(new String[0])),
+                            RestStatus.CONFLICT
+                        )
+                    );
+
+            }
+
+        }, e -> {
+            log.error("Failed to delete ML Model: " + modelId, e);
+            actionListener.onFailure(e);
+
+        }));
+
+    }
+
+    private void checkSearchPipelineBeforeDeleteModel(String modelId, ActionListener<Boolean> actionListener) {
+        GetSearchPipelineRequest getSearchPipelineRequest = new GetSearchPipelineRequest();
+        client.execute(GetSearchPipelineAction.INSTANCE, getSearchPipelineRequest, ActionListener.wrap(searchPipelineResponse -> {
+            if (!isPipelineContainsModel(
+                searchPipelineResponse.pipelines(),
+                modelId,
+                org.opensearch.search.pipeline.PipelineConfiguration::getConfigAsMap
+            )) {
+                actionListener.onResponse(true);
+            } else {
+                List<String> searchPipelineIds = getAllPipelineIds(
+                    searchPipelineResponse.pipelines(),
+                    org.opensearch.search.pipeline.PipelineConfiguration::getId
+                );
+                actionListener
+                    .onFailure(
+                        new OpenSearchStatusException(
+                            searchPipelineIds.size()
+                                + " search pipelines are still using this model, please delete or update the pipelines first: "
```

Comment:
nit: 1 search pipeline.
Also keep checkAgentBeforeDeleteModel and checkIngestPipelineBeforeDeleteModel consistent.